src/clients/c++/perf_analyzer/client_backend/torchserve/http_client.h

```diff
@@ -108,18 +114,30 @@ class HttpClient : public nic::InferenceServerClient {
 
 class HttpInferRequest {
  public:
-  HttpInferRequest() : header_list_(nullptr) {}
+  struct Deleter {
+    void operator()(FILE* file)
+    {
+      // Do nothing
+    }
+  };
+
+  HttpInferRequest();
   ~HttpInferRequest();
   Error InitializeRequest();
+  Error OpenFileData(std::string& file_path);
+  long FileSize();
+  Error CloseFileData();
   nic::RequestTimers& Timer() { return timer_; }
   std::string& DebugString() { return *infer_response_buffer_; }
+  FILE* FilePtr() { return file_ptr_.get(); }
   friend HttpClient;
   friend InferResult;
 
  private:
   // Pointer to the list of the HTTP request header, keep it such that it will
   // be valid during the transfer and can be freed once transfer is completed.
   struct curl_slist* header_list_;
+  std::unique_ptr<FILE, Deleter> file_ptr_;
```

Comment:
Why you provide a deleter if it does nothing?

Reply:
The default deleter for unique_ptr will call delete on the FILE*. But, the close call is already clearing off the resources. If I don't provide an empty deleter a second attempt to delete the pointer will be made when the unique_ptr object is destructed. An alternative is to check and close the FILE* in the deleter instead of [here](https://github.com/triton-inference-server/server/blob/1398887ecfc686bb4703cdbb04c487d7fcef75e0/src/clients/c%2B%2B/perf_analyzer/client_backend/torchserve/http_client.cc#L182). But I would prefer to close the file as soon as we are done uploading the contents.

Reply:
If you are using an explicit function for closing the file, then I don't think you really need a smart pointer. Besides that, it is possible to have resource leak if `CloseFileData()` doesn't get called, smart pointer with empty deleter doesn't help that either. I would have moved the closing file logic inside the deleter and explicitly call `reset()` if you want to close it earlier (or let `CloseFileData()` be just calling reset() to make the intention more clear).